### DISCUSSION TITLE
"Postal inspects all incoming messages for spam and other threats"?

### DISCUSSION THREAD

**Discussion by JohnHammell** [postalserver/postal#3177] [Help with using Postal] [answered]
Saw the following in my Postal settings in the UI:
>"Postal inspects all incoming messages for spam and other threats. Incoming messages are assigned a score which represents how likely an e-mail is to be spam. From here you can choose at which level you'd like to identify messages as spam."

How exactly does Postal do this scanning? Is this by default?

Does Postal check DKIM & SPF on inbound email & reject any messages that fail?

I saw rspamd, spamd & clamav in yaml.yml at https://github.com/postalserver/postal/blob/main/doc/config/yaml.yml but these settings were not in my default postal.yml upon install.

Would like to get a better understanding of how "Postal inspects all incoming messages for spam and other threats" so that I can make sure I have it all configured properly, etc. Also would like to make sure DKIM & SPF are being checked to help block potential domain spoofing on inbound email.

**Comment by willpower232** [comment, marked as answer]
The wording is probably a bit confusing, its been a while since I looked at it.

> How exactly does Postal do this scanning? Is this by default?

> I saw rspamd, spamd & clamav in yaml.yml

You have to install whichever services you like on your server and can then fill in the settings in the config file (or environment variables), Postal does not install them for you but will work which whichever you set up, scanning messages as you set it up to.

> Does Postal check DKIM & SPF on inbound email & reject any messages that fail?

I don't believe Postal does itself however I'd hope that (r)spamd checked those things.
